src/map/script.c

```diff
@@ -23880,11 +23886,7 @@ static BUILDIN(pushpc)
 		cells = -cells;
 	}
 
-	Assert_retr(false, dir >= UNIT_DIR_FIRST && dir < UNIT_DIR_MAX);
-	dx = dirx[dir];
-	dy = diry[dir];
-
-	unit->blown(&sd->bl, dx, dy, cells, 0);
+	unit->blown(&sd->bl, dir, cells, 0);
```

Comment:
You've changed the data type of the last parameter of `unit_blown()` to `bool`, so passing 0 is invalid.

Reply:
Good catch!